### DISCUSSION TITLE
Unable to download soil data

### DISCUSSION THREAD

**Discussion by ParameshwariS-tiger** [APSIMInitiative/ApsimX#8864] [Q&A] [answered]
Hi all, I was able to download soil data for the below coordinates few days back but now I am unable to do so. It says "0 soils found"
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/54f41582-cabe-478b-a8ee-014f8d7808c6)

Below image is the downloaded soil for the same few days back
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/eed79ba5-ecf6-4514-8f34-817077297242)

I tried with different other coordinates for which I have downloaded before but now I am facing this problem

Any assistance on this would be greatly appreciated, thanks:)

**Comment by hol353** [comment]
Looks like a defect. Converting this to an Issue.

**Reply by ParameshwariS-tiger** [reply]
@hol353 Sure, thanks

**Reply by ParameshwariS-tiger** [reply]
Facing this issue again, I was able to download few days back but not able to do now. Kindly check

**Comment by ric394** [comment]
Looks to be working again this morning.

**Reply by ParameshwariS-tiger** [reply]
Yes, it is working now thanks :)

**Comment by ParameshwariS-tiger** [comment]
Unable to download the soil again, now it says "Object reference not set to an instance of an object."

**Reply by KeLiu7** [reply]
Still not working properly..

**Reply by ric394** [reply]
A new build was not created for the fix that was submitted recently. I'll have a new build for this shortly.

**Reply by KeLiu7** [reply]
> A new build was not created for the fix that was submitted recently. I'll have a new build for this shortly.

That's great rich - Previously users could only download soil data from ISRIC. Will this update include APSOIL as well?

**Reply by ric394** [reply, marked as answer]
Running the fixes locally allows soils from all sources as shown here:
![image](https://github.com/user-attachments/assets/0bb28c24-7814-4bf3-8735-ce9efc29ac26)

**Reply by ParameshwariS-tiger** [reply]
Is the issue resolved? I am not able to download soils still

**Reply by ParameshwariS-tiger** [reply]
But still it says "Object reference not set to an instance of an object." for me

**Comment by ParameshwariS-tiger** [comment]
Hi, any update on this issue? Still facing the same problem

**Reply by jsarkercqu** [reply]
Hi @ParameshwariS-tiger, Did you get a solution to your issue? I can't download soil data either.

**Reply by ric394** [reply]
@ParameshwariS-tiger  @jsarkercqu Are you able to download soils at the moment?

**Reply by ric394** [reply]
Also, if you have an error message that would helpful as well.

**Reply by ParameshwariS-tiger** [reply]
@jsarkercqu @ric394 It's working fine for me, able to download soils

**Reply by jsarkercqu** [reply]
@ParameshwariS-tiger, Thanks for your reply. Can you please advise how did you download this? I can't see any option.

**Reply by ParameshwariS-tiger** [reply]
The same approach, right click on paddock/field, select "Download Soils", and enter the details. Then you can select the soils and add to your simulation.

**Reply by jsarkercqu** [reply]
@ParameshwariS-tiger, Thanks a lot. Much appreciated.
